src/openmrs-resource/visit.resource.ts

```diff
@@ -0,0 +1,69 @@
+import { openmrsObservableFetch } from "@openmrs/esm-api";
+
+import { Observable } from "rxjs";
+import { take } from "rxjs/operators";
+import { FetchResponse } from "@openmrs/esm-api/dist/openmrs-fetch";
+
+export function getVisitsForPatient(
+  patientUuid: string,
+  abortController: AbortController,
+  v?: string
+): Observable<FetchResponse<any>> {
+  const custom =
+    v ||
+    "custom:(uuid,encounters:(uuid,encounterDatetime," +
+      "form:(uuid,name),location:ref," +
+      "encounterType:ref,encounterProviders:(uuid,display," +
+      "provider:(uuid,display))),patient:(uuid,uuid)," +
+      "visitType:(uuid,name),attributes:(uuid,display,value),location:ref,startDatetime," +
+      "stopDatetime)";
+
+  return openmrsObservableFetch(
+    `/ws/rest/v1/visit?patient=${patientUuid}&v=${custom}`,
+    {
+      signal: abortController.signal,
+      method: "GET",
+      headers: {
+        "Content-type": "application/json"
+      }
+    }
+  ).pipe(take(1));
+}
+
+export function saveVisit(
+  payload: NewVisitPayload,
+  abortController: AbortController
+): Observable<FetchResponse<any>> {
+  return openmrsObservableFetch(`/ws/rest/v1/visit`, {
```

Comment:
Is there an advantage of using an observable over a promise? I realized that in most cases we have used promises with ` createErrorHandler()` on the component to handle errors.

Reply:
The is not really a preference of one over the other. For this particular repo, we've used Observable fetch a lot. Would be inconsistent to introduce it now, because it will need refactors in all places using Observable fetch.  But then again, RXjs is rich in operators that might make manipulation much easier.